src/utils/permissions.ts

```diff
@@ -1,117 +1,69 @@
-) {
+import { Role } from "../structures/structures.ts";
+import {
+  Errors,
+  Permission,
+  Permissions,
+  RawOverwrite,
+} from "../types/types.ts";
+
+/** Returns you the permission bits from the member in this guild */
+export async function calculateServerPerm(memberID: string, guildID: string) {
   const guild = await cacheHandlers.get("guilds", guildID);
-  if (!guild) return false;
-
-  if (memberID === guild.ownerID) return true;
-
-  const member = (await cacheHandlers.get("members", memberID))?.guilds.get(
-    guildID,
-  );
-  if (!member) return false;
-
-  return memberHasPermission(memberID, guild, member.roles, permissions);
-}
-
-/** Checks if the member has this permission. If the member is an owner or has admin perms it will always be true. */
-export function memberHasPermission(
-  memberID: string,
-  guild: Guild,
-  memberRoleIDs: string[],
-  permissions: Permission[],
-) {
-  if (memberID === guild.ownerID) return true;
-
-  const permissionBits = memberRoleIDs.map((id) =>
-    guild.roles.get(id)?.permissions
-  )
-    // Removes any edge case undefined
-    .filter((id) => id)
-    .reduce((bits, permissions) => {
-      bits |= BigInt(permissions);
-      return bits;
-    }, BigInt(0));
+  if (!guild) throw Error(Errors.GUILD_NOT_FOUND);
 
-  if (permissionBits & BigInt(Permissions.ADMINISTRATOR)) return true;
-
-  return permissions.every((permission) =>
-    permissionBits & BigInt(Permissions[permission])
-  );
-}
-
-export async function botHasPermission(
-  guildID: string,
-  permissions: Permission[],
-) {
-  const guild = await cacheHandlers.get("guilds", guildID);
-  if (!guild) return false;
-
-  // Check if the bot is the owner of the guild, if it is, returns true
-  if (guild.ownerID === botID) return true;
+  // Check if the member is the owner of the guild, if he is, returns admin permissions
+  if (memberID === guild.ownerID) return "8";
 
   const member = (await cacheHandlers.get("members", botID))?.guilds.get(
     guildID,
   );
-  if (!member) return false;
+  if (!member) throw Error(Errors.MEMBER_NOT_FOUND);
 
-  // The everyone role is not in member.roles
-  const permissionBits = [...member.roles, guild.id]
+  return [...member.roles, guild.id]
     .map((id) => guild.roles.get(id)!)
     // Remove any edge case undefined
-    .filter((r) => r)
+    .filter((role) => role)
     .reduce((bits, data) => {
       bits |= BigInt(data.permissions);
 
       return bits;
-    }, BigInt(0));
-
-  if (permissionBits & BigInt(Permissions.ADMINISTRATOR)) return true;
-
-  return permissions.every((permission) =>
-    permissionBits & BigInt(Permissions[permission])
-  );
+    }, BigInt(0)).toString();
 }
 
-/** Checks if the bot has the permissions in a channel */
-export function botHasChannelPermissions(
-  channelID: string,
+/** Checks if the member has this permission. If the member is an owner or has admin perms it will always be true. */
+export async function hasServerPerm(
```

Comment:
Can you rename this to `hasServerPermissions()`?
Also, could you change the position of the parameters? `guildID` being the first parameter and `memberID` the second.